Gruntfile.js

```diff
@@ -50,6 +50,24 @@ const applyPortShift = (env) => {
   return { ...env, API_PORT, API_URL }
 }
 
+const untilServerIsReady = () => {
+  console.log('waiting for the server to be ready')
+  // Wait for the server to be ready.
+  const t0 = Date.now()
+  const timeout = 30000
+  return new Promise((resolve, reject) => {
+    (function ping () {
+      fetch(process.env.API_URL).then(resolve).catch(() => {
```

Comment:
I think you can [directly cause `fetch` to timeout](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#aborting_a_fetch):

![Screen Shot 2023-04-29 at 3 26 17 PM](https://user-images.githubusercontent.com/138706/235326594-faf67b96-8d52-4b22-9897-bbbe2cac9732.jpg)

Reply:
This technique could be useful in the client, for example to implement a button allowing users to stop fetching a large ressource when it takes too long.

In this function, the specified timeout is not meant for individual fetch requests, but is the total time we're willing to spend polling the server before giving up. That's why it's quite generous (30s).
Aborting individual requests would defeat their purpose, which is to test whether the server is ready. The ressource is not used.
If the server is not up, then any fetch quickly fails by itself. According to my tests it won't stay pending waiting to be aborted. So In that case the `catch()` clause runs and schedules another fetch request, unless the timeout has been exceeded.

Reply:
Ah, I have a better understanding of how this function works now, thanks, resolving this